website/for_teachers.py

```diff
@@ -184,13 +185,137 @@ def get_class(self, user, class_id):
 
     @route("/customize-class/<class_id>", methods=["GET"])
     @requires_login
-    def get_class_info(self, user, class_id):
+    def get_class_customization_page(self, user, class_id):
         if not is_teacher(user) and not is_admin(user):
             return utils.error_page(error=403, ui_message=gettext("retrieve_class_error"))
         Class = self.db.get_class(class_id)
         if not Class or (Class["teacher"] != user["username"] and not is_admin(user)):
             return utils.error_page(error=404, ui_message=gettext("no_such_class"))
 
+        session['class_id'] = class_id
+        customizations, adventure_names, available_adventures, min_level = self.get_class_info(user, class_id)
+
+        return render_template(
+            "customize-class.html",
+            page_title=gettext("title_customize-class"),
+            class_info={"name": Class["name"], "id": Class["id"], "teacher": Class["teacher"]},
+            max_level=hedy.HEDY_MAX_LEVEL,
+            customizations=customizations,
+            adventure_names=adventure_names,
+            available_adventures=available_adventures,
+            adventures_default_order=hedy_content.ADVENTURE_ORDER_PER_LEVEL,
+            current_page="for-teachers",
+            min_level=min_level,
+            class_id=class_id,
+            javascript_page_options=dict(
+                page='customize-class',
+                class_id=class_id
+            ))
+
+    @route("/change-level", methods=["GET"])
+    @requires_login
+    def change_dropdown_level(self, user):
+        if not is_teacher(user) and not is_admin(user):
+            return utils.error_page(error=403, ui_message=gettext("retrieve_class_error"))
+        Class = self.db.get_class(session['class_id'])
+        if not Class or (Class["teacher"] != user["username"] and not is_admin(user)):
+            return utils.error_page(error=404, ui_message=gettext("no_such_class"))
+
+        level = request.args.get('level')
+        customizations, adventure_names, available_adventures, _ = self.get_class_info(user, session['class_id'])
+
+        return render_partial('customize-class/sortable-adventures.html',
```

Comment:
Does `render_partial` here do anything different than `render_template` ?

Reply:
Sorry for not commenting this yesterday! 

We discussed it in the last Hedy meeting, basically this function works like a macro or an include, in the sense that you can include this template inside another template; but neither macro nor include can be called from python, you can only call `render_template`, `render_partial` can be called so its a bit of a combination. Another justification, that is crutial, is that when called inside a template, you can pass values to the partial template. The author justifies the existence of this library here: https://github.com/mikeckennedy/jinja_partials/issues/1 and I found it in this talk that he gave: https://www.youtube.com/watch?v=kNXVFB5eQfo

I think that this is a good library moving forward, because makes working with Flask and HTMX much more easier and flexible. Also, don't seem to be a library that can brake randomly due to an update, as it seems that development finished. I'm also very willing to listen to your comments in the use of this library.

Reply:
Thanks for the explanation!! We should also move this to the docs, will you remember this or should I make an issue?

Reply:
I’m still a little confused. Don’t let that bother you or anything, it’s not a reason to block anything on. Let’s get this majestic piece of work merged!

But here’s where I’m at:

- I see how `render_partial` is nicer in a template than `{% include %}`
- I don’t see how `render_partial` is different from `render_template` in Python code
- I don’t see how `render_partial` helps with macros called from Python.